internal/resources/azure/storage_account.go

```diff
@@ -0,0 +1,911 @@
+		return nil
+	}
+
+	if r.isPremium() {
+		// Premium tier doesn't differentiate between Hot or Cool storage. This
+		// helps to simplify skuName search.
+		r.AccessTier = "Premium"
+	}
+
+	costComponents := []*schema.CostComponent{}
+
+	costComponents = append(costComponents, r.storageCostComponents()...)
+
+	costComponents = append(costComponents, r.dataAtRestCostComponents()...)
+	costComponents = append(costComponents, r.snapshotsCostComponents()...)
+	costComponents = append(costComponents, r.metadataAtRestCostComponents()...)
+
+	costComponents = append(costComponents, r.iterativeWriteOperationsCostComponents()...)
+	costComponents = append(costComponents, r.writeOperationsCostComponents()...)
+	costComponents = append(costComponents, r.listAndCreateContainerOperationsCostComponents()...)
+	costComponents = append(costComponents, r.iterativeReadOperationsCostComponents()...)
+	costComponents = append(costComponents, r.readOperationsCostComponents()...)
+	costComponents = append(costComponents, r.otherOperationsCostComponents()...)
+	costComponents = append(costComponents, r.dataRetrievalCostComponents()...)
+	costComponents = append(costComponents, r.dataWriteCostComponents()...)
+	costComponents = append(costComponents, r.blobIndexTagsCostComponents()...)
+
+	costComponents = append(costComponents, r.earlyDeletionCostComponents()...)
+
+	return &schema.Resource{
+		Name:           r.Address,
+		UsageSchema:    StorageAccountUsageSchema,
+		CostComponents: costComponents,
+	}
+}
+
+// buildProductFilter returns a product filter for the Storage Account's products.
+func (r *StorageAccount) buildProductFilter(meterName string) *schema.ProductFilter {
+	var productName string
+
+	switch {
+	case r.isBlockBlobStorage():
+		productName = map[string]string{
+			"Standard": "Blob Storage",
+			"Premium":  "Premium Block Blob",
+		}[r.AccountTier]
+	case r.isStorageV2():
+		if r.NfsV3 {
+			productName = map[string]string{
+				"Standard": "General Block Blob v2 Hierarchical Namespace",
+				"Premium":  "Premium Block Blob v2 Hierarchical Namespace",
+			}[r.AccountTier]
+		} else {
+			productName = map[string]string{
+				"Standard": "General Block Blob v2",
+				"Premium":  "Premium Block Blob",
+			}[r.AccountTier]
+		}
+	case r.isFileStorage():
+		productName = map[string]string{
+			"Standard": "Files v2",
+			"Premium":  "Premium Files",
+		}[r.AccountTier]
+	}
+
+	skuName := fmt.Sprintf("%s %s", r.AccessTier, r.AccountReplicationType)
+
+	return &schema.ProductFilter{
+		VendorName:    strPtr("azure"),
+		Region:        strPtr(r.Region),
+		Service:       strPtr("Storage"),
+		ProductFamily: strPtr("Storage"),
+		AttributeFilters: []*schema.AttributeFilter{
+			{Key: "productName", Value: strPtr(productName)},
+			{Key: "skuName", Value: strPtr(skuName)},
+			{Key: "meterName", ValueRegex: regexPtr(fmt.Sprintf("%s$", meterName))},
```

Comment:
@alikhajeh1 As we discussed before, here I tried this helper function `regexPtr` to ensure the case-insensitive search. What do you think, is it intuitive to use?
```go
func regexPtr(regex string) *string {
	return strPtr(fmt.Sprintf("/%s/i", regex))
}
```

Reply:
I saw that before and thought nice! :)